test/e2e/common/autoscaling_utils.go

```diff
@@ -333,26 +334,22 @@ func (rc *ResourceConsumer) GetReplicas() int {
 
 func (rc *ResourceConsumer) WaitForReplicas(desiredReplicas int) {
 	timeout := 15 * time.Minute
-	for start := time.Now(); time.Since(start) < timeout; time.Sleep(20 * time.Second) {
-		if desiredReplicas == rc.GetReplicas() {
-			framework.Logf("%s: current replicas number is equal to desired replicas number: %d", rc.kind, desiredReplicas)
-			return
-		} else {
-			framework.Logf("%s: current replicas number %d waiting to be %d", rc.kind, rc.GetReplicas(), desiredReplicas)
-		}
-	}
-	framework.Failf("timeout waiting %v for pods size to be %d", timeout, desiredReplicas)
+	pollingInterval := 20 * time.Second
+	EventuallyWithOffset(1, func() int {
```

Comment:
Instead of this, I would prefer to use the thing that we are using all over the code, which is:
"wait.PollImmediate()"

Consistency across codebase is pretty important in my opinion.